src/HooglePlus/GHCChecker.hs

```diff
@@ -121,6 +124,37 @@ checkStrictness tyclassCount body sig modules =
         (\(SomeException _) -> return False)
         (checkStrictness' tyclassCount body sig modules)
 
+check :: Goal -> SearchParams -> Chan Message -> Chan Message -> IO ()
+check goal searchParams solverChan checkerChan = catch
+    (evalStateT (check_ goal searchParams solverChan checkerChan) emptyFilterState)
+    (\err ->
+        writeChan checkerChan (MesgLog 0 "error" (show err)) >>
```

Comment:
The tag on there is to help someone locate the problem. Instead of `error`, which could appear in a couple dozen places in the codebase, try a more localized and descriptive name like `filterCheck` and prepend `error:` to the actual message.